modules/installation-azure-service-principal.adoc

```diff
@@ -0,0 +1,124 @@
+.. To view your account details, run the following command:
++
+----
+$ az account show
+{
+  "environmentName": "AzureCloud",
+  "id": "9bab1460-96d5-40b3-a78e-17b15e978a80",
+  "isDefault": true,
+  "name": "Subscription Name", <1>
+  "state": "Enabled",
+  "tenantId": "6057c7e9-b3ae-489d-a54e-de3f6bf6a8ee",
+  "user": {
+    "name": "you@example.com",
+    "type": "user"
+  }
+}
+----
+<1> Ensure that the value of the name parameter is the name of your subscription.
+
+.. You can change the subscription to use by obtaining the subscription ID
+from the Azure portal and setting it in the CLI:
++
+----
+$ az account set -s <subscription_ID>
+----
+
+.. If you changed the subscription ID, display your account information again:
++
+----
+$ az account show
+
+{
+  "environmentName": "AzureCloud",
+  "id": "33212d16-bdf6-45cb-b038-f6565b61edda",
+  "isDefault": true,
+  "name": "Subscription Name",
+  "state": "Enabled",
+  "tenantId": "8049c7e9-c3de-762d-a54e-dc3f6be6a7ee",
+  "user": {
+    "name": "you@example.com",
+    "type": "user"
+  }
+}
+----
+
+. Record the values of the `tenantId` and `id` parameters from the previous
+output. You need these values during {product-title} installation.
+
+. Create the service principal for your account:
++
+----
+$ az ad sp create-for-rbac --role "User Access Administrator" -role Contributor --name <service_principal> <1>
+Changing "<service_principal>" to a valid URI of "http://<service_principal>", which is the required format used for service principal names
+Retrying role assignment creation: 1/36
+Retrying role assignment creation: 2/36
+Retrying role assignment creation: 3/36
+Retrying role assignment creation: 4/36
+{
+  "appId": "8bd0d04d-0ac2-43a8-928d-705c598c6956",
+  "displayName": "<service_principal>",
+  "name": "http://<service_principal>",
+  "password": "ac461d78-bf4b-4387-ad16-7e32e328aec6",
+  "tenant": "6048c7e9-b2ad-488d-a54e-dc3f6be6a7ee"
+}
+
+----
+<1> Specify the name to assign to the service principal.
+
+. Record the values of the `appId` and `password` parameters from the previous
+output. You need these values during {product-title} installation.
+
+. Grant additional permissions to the service principal. The service principal
+requires the legacy `Azure Active Directory Graph -> Application.ReadWrite.OwnedBy`
+permission and the `User Access Administrator` role for the cluster to
+credentials for its components.
+.. To assign the `User Access Administrator` role, run the following command:
++
+----
+$  az role assignment create --role "User Access Administrator" \
+    --assignee-object-id $(az ad sp list --filter "appId eq '<appId>'" \ <1>
+       | jq '.[0].objectId' -r)
+----
+<1> Specify the `appId` parameter value for your service principal.
+
+.. To assign the `Azure Active Directory Graph` permission, run the following
+command:
++
+----
+$ az ad app permission add --id <appId> \ <1>
```

Comment:
These very specific values make me a little nervous; have we tried this outside of the `jamesrussell1911` tenancy? Could we try in the `rhcuppett` one, since that's definitely independent?

Reply:
We may reference https://blogs.msdn.microsoft.com/aaddevsup/2018/06/06/guid-table-for-windows-azure-active-directory-permissions/ to clarify on why such specific values are used.

Reply:
I'm happy to  add a link to that table.

Who has access to that account to try it out?

Reply:
Because of the info in the linked webpage, I'll withdraw my concern that the information might be different for different tenancies; I don't think we need to try it out on a different tenancy.

However I'll note that the linked webpage has a prominent notice at the top:

> Note: That these GUIDs are subject to change in the future and may not be the same anymore.

Reply:
Stephen gave me access to his other account, but I'm having trouble making a service principal. He says that he granted me the “Application developer” role, but I still see "Resource access" in the Azure portal.
@coverprice, do you have any thoughts or suggestions? (I want to be sure that we have an accurate description of the required permissions for the docs!)